src/test/java/BoardTest.java

```diff
@@ -243,4 +243,109 @@ public void testMovePawnsIllegally() throws InvalidPositionException
 		/*16 b/c there are 16 pawns that get moved illegally*/
 		assertEquals(numErrors, 16);
 	}
+	@Test
+	public void testValidMoveToNewPosition() 
+	{
+		
+		Board board = new Board();
+		/* Moving pawns out of the way for tests of other pieces */
+		try
+		{
+			board.movePiece(new Position(1, 0), new Position(3, 0));
+			board.movePiece(new Position(1,1), new Position(3,1));
+			board.movePiece(new Position(1,5), new Position(3,5));
+			board.movePiece(new Position(0,0), new Position(2,0));
+			board.movePiece(new Position(0,1), new Position(2,2));
+			board.movePiece(new Position(0,2), new Position(1,1));
+			board.movePiece(new Position(0,4), new Position(3,7));
+			board.movePiece(new Position(0,3), new Position(0,4));
+			
+		}
+		catch(InvalidPositionException ex)
+		{
+			fail(ex.getLocalizedMessage());
+		}
+	}
+	@Test 
+	public void testInvalidMoveToOutsideGrid()
+	{
+		Board board = new Board();
+		try
+		{
+			int row =0;
+			for(int col=0; col < 8; col++)
+			{
+				board.movePiece(new Position(row, col), new Position(row, col-1));
+			}
+			board.movePiece(new Position(0,0), new Position(-1,0));
+			board.movePiece(new Position(1,0), new Position(-2,0));
+			board.movePiece(new Position(0,7), new Position(0,8));
+			board.movePiece(new Position(1,7), new Position(1,8));
+		}
+		catch(InvalidPositionException ex)
+		{
+			fail(ex.getLocalizedMessage());
+		}
+	}
+	@Test 
+	public void testInvalidMoveOntoOtherPieces()
```

Comment:
- From my understanding from looking at it, this test is going to fail. We are expecting exceptions to be thrown from moving the pieces onto other pieces, but the code does not specify that it is expecting the exceptions to be thrown. I don't think I explained this clearly, so we should meet up ad discuss this part.